test/e2e/validation_test.go

```diff
@@ -0,0 +1,188 @@
+	"github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachineconfig/tracker"
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/ssh"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+)
+
+// waitForTrackerConfigMap to be created waits for the Windows tracker configmap to be created with appropriate values
+func waitForTrackerConfigMap(kubeclient kubernetes.Interface, namespace string, expectedNodesToBeTracked int,
+	retryInterval, timeout time.Duration) error {
+	err := wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		trackerConfigMap, err := kubeclient.CoreV1().ConfigMaps(namespace).Get(tracker.StoreName, metav1.GetOptions{})
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				log.Printf("Waiting for availability of tracker configmap to be created: %s\n", tracker.StoreName)
+				return false, nil
+			}
+			return false, err
+		}
+		if len(trackerConfigMap.BinaryData) == expectedNodesToBeTracked {
+			log.Println("Tracker configmap tracking required number of configmap")
+			return true, nil
+		}
+		log.Printf("still waiting for %d number of "+
+			"Windows worker nodes to be tracked but as of now we have %d\n", expectedNodesToBeTracked,
+			len(trackerConfigMap.BinaryData))
+		return false, nil
+	})
+	return err
+}
+
+// getInstanceID gets the instanceID of VM for a given cloud provider ID
+// Ex: aws:///us-east-1e/i-078285fdadccb2eaa. We always want the last entry which is the instanceID
+func getInstanceID(providerID string) string {
+	providerTokens := strings.Split(providerID, "/")
+	return providerTokens[len(providerTokens)-1]
+}
+
+// getInstanceIDsCreated returns the instanceIDs of all the Windows nodes created
+func getInstanceIDsCreated(kubeclient kubernetes.Interface) ([]string, error) {
+	nodes, err := kubeclient.CoreV1().Nodes().List(metav1.ListOptions{LabelSelector: wmc.WindowsOSLabel})
+	if err != nil {
+		return nil, errors.Wrap(err, "error while querying for Windows nodes")
+	}
+	instanceIDs := make([]string, 0, len(nodes.Items))
+	for _, node := range nodes.Items {
+		if len(node.Spec.ProviderID) > 0 {
+			instanceID := getInstanceID(node.Spec.ProviderID)
+			instanceIDs = append(instanceIDs, instanceID)
+		}
+	}
+	return instanceIDs, nil
+}
+
+// testConfigMapValidation ensures that the required configmap is created and is having appropriate
+// entries
+func testConfigMapValidation(t *testing.T, nodeCount int) {
+	testCtx := framework.NewTestCtx(t)
+	namespace, err := testCtx.GetNamespace()
+	require.NoError(t, err, "error while getting test namespace")
+
+	err = waitForTrackerConfigMap(framework.Global.KubeClient, namespace, nodeCount,
```

Comment:
Is there a reason for passing `framework.Global.KubeClient` around? It seems like an unnecessary input. You either get it from the SDK framework or have a local struct around it.

Reply:
I want to have it in a struct and have the struct passed around to every test suite. Instead of nodeCount this struct will be passed around. I will address it later.

Reply:
Where is it being tracked?

By doing this you are creating a pattern that everyone else will follow and increasing the amount of nice to have work which will never get addressed.